Utilities/Mergers/test/vectorTopologyCommon.h

```diff
@@ -0,0 +1,162 @@
+      DataProcessorSpec producer{
+        "producer-vec" + std::to_string(producerIdx),
+        Inputs{},
+        Outputs{{{"mo"}, origin, description, static_cast<SubSpecificationType>(producerIdx), Lifetime::Sporadic}},
+        AlgorithmSpec{static_cast<AlgorithmSpec::ProcessCallback>([producerIdx, numberOfProducers, binsCount = mHistoBinsCount, histoMin = mHistoMin, histoMax = mHistoMax, sent = false](ProcessingContext& processingContext) mutable {
+          if (sent) {
+            std::this_thread::sleep_for(std::chrono::milliseconds{100});
+            return;
+          }
+
+          const auto subspec = static_cast<SubSpecificationType>(producerIdx);
+          auto vectorOfHistos = std::make_unique<mergers::VectorOfRawTObjects>(2);
+
+          int i = 0;
+          for (auto& hist_ptr : *vectorOfHistos) {
+            const auto histoname = std::string{"histo"} + std::to_string(++i);
+            auto* hist = new TH1F(histoname.c_str(), histoname.c_str(), binsCount, histoMin, histoMax);
+            hist->Fill(producerIdx);
+            hist->Fill(5);
+            hist_ptr = hist;
+          }
+
+          processingContext.outputs().snapshot(OutputRef{"mo", subspec}, *vectorOfHistos);
+          for_each(vectorOfHistos->begin(), vectorOfHistos->end(), [](auto& histoPtr) { delete histoPtr; });
+          sent = true;
+        })}};
+      specs.push_back(producer);
+    }
+    return mergersInputs;
+  }
+
+  void generateMergers(WorkflowSpec& specs, const Inputs& mergersInputs, mergers::InputObjectsTimespan mergerType)
+  {
+    using namespace mergers;
+
+    MergerInfrastructureBuilder mergersBuilder;
+    mergersBuilder.setInfrastructureName("vec");
+    mergersBuilder.setInputSpecs(mergersInputs);
+    mergersBuilder.setOutputSpec({{"main"}, origin, description, 0});
+    MergerConfig config;
+    config.inputObjectTimespan = {mergerType};
+    std::vector<std::pair<size_t, size_t>> param = {{5, 1}};
+    config.publicationDecision = {PublicationDecision::EachNSeconds, param};
+    config.mergedObjectTimespan = {MergedObjectTimespan::FullHistory};
+    config.topologySize = {TopologySize::NumberOfLayers, 2};
+    mergersBuilder.setConfig(config);
+
+    mergersBuilder.generateInfrastructure(specs);
+  }
+
+  void generateChecker(WorkflowSpec& specs)
+  {
+    specs.push_back(DataProcessorSpec{
+      "data-checker",
+      Inputs{{"vec", origin, description, 0, Lifetime::Sporadic}},
+      Outputs{},
+      AlgorithmSpec{
+        AlgorithmSpec::InitCallback{[expectedResult = mExpectedResult](InitContext&) {
+          // reason for this crude retry is that multiple layers are not synchronized between each other and publish on their own timers
+          return AlgorithmSpec::ProcessCallback{[expectedResult, retryNumber = 0, retries = 5](ProcessingContext& processingContext) mutable {
+            if (retryNumber++ == retries) {
```

Comment:
if this is first, before the check, it basically means it will fail only at the 6th time we receive an object. If the test times out before it, we will miss why the test failed (no correct data at 5th time). Perhaps consider rearranging it, if it is not a lot of effort.